### PR TITLE
plan: process the case that index col has specified length.

### PR DESCRIPTION
**Pull request by hanfei1991**
@shenli @coocood @zimulala @winoros  PTAL

### CODE REVIEW

File: plan/logical_plan_test.go [hunk 1]
```diff
@@ -158,6 +158,27 @@ func mockResolve(node ast.Node) (infoschema.InfoSchema, error) {
 			},
 			State: model.StatePublic,
 		},
+		{
+			Name: model.NewCIStr("e_d_c_str"),
```
Comment (coocood) [inline plan/logical_plan_test.go]: It's hard to read the test code if we can't tell from the index name that `c` is a prefix column

Comment (hanfei1991) [inline plan/logical_plan_test.go]: any suggestion?

Comment (coocood) [inline plan/logical_plan_test.go]: how about `e_d_c_str_prefix`?

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**